src/DocumentationBrowserViewExtension/DocumentationBrowserViewExtension.cs

```diff
@@ -14,7 +14,7 @@ namespace Dynamo.DocumentationBrowser
     /// The DocumentationBrowser view extension displays web or local html files on the Dynamo right panel.
     /// It reacts to documentation display request events in Dynamo to know what and when to display documentation.
     /// </summary>
-    public class DocumentationBrowserViewExtension : IViewExtension, ILogSource
+    public class DocumentationBrowserViewExtension : IViewExtensionBaseClass, IViewExtension, ILogSource
```

Comment:
Extending a new class to a public class wouldn't be a breaking change, right?

Reply:
Should be fine, I think that's how we used to do it. @mjkkirschner Please confirm?

Reply:
`IViewExtensionBaseClass` is a strange name the `I` usually signifies Interface - I think this is against c# convention.

Reply:
hmmm- it doesn't sound good...

❓ REQUIRES JUDGMENT: Introducing a new base class

A type can be introduced into a hierarchy between two existing types if it doesn't introduce any new abstract members or change the semantics or behavior of existing types. For example, in .NET Framework 2.0, the DbConnection class became a new base class for SqlConnection, which had previously derived directly from Component.


https://docs.microsoft.com/en-us/dotnet/core/compatibility/

we'll have to do some reading to understand why - @mmisol any thoughts?

Reply:
ahh, I think I understand - if we do this - we should provide empty default implementations for all methods in the abstract base class, or other extensions that derive from our extensions will fail to implement that abstract class correctly.

On the other hand - I don't know if we consider our Extension classes as part of our API - that would be a great thing to document somewhere.

Reply:
Yes, changed the name to ViewExtensionBaseClass. This has empty default implementations for all the methods from the interface.